webapp/channels/src/components/admin_console/schema_text.tsx

```diff
@@ -3,73 +3,55 @@
 
 import marked from 'marked';
 import React from 'react';
+import type {MessageDescriptor} from 'react-intl';
 import {FormattedMessage} from 'react-intl';
 
 import FormattedMarkdownMessage, {CustomRenderer} from 'components/formatted_markdown_message';
 
 type Props = {
     isMarkdown?: boolean;
-    isTranslated?: boolean;
-    text: string | object;
-    textDefault?: string;
+    text: string | MessageDescriptor | JSX.Element;
     textValues?: Record<string, React.ReactNode>;
 }
 
-export default class SchemaText extends React.PureComponent<Props> {
-    static defaultProps = {
-        isTranslated: true,
-    };
-
-    renderTranslated = () => {
-        const {
-            isMarkdown,
-            text,
-            textDefault,
-            textValues,
-        } = this.props;
+const SchemaText = ({
+    isMarkdown,
+    text,
+    textValues,
+}: Props) => {
+    if (typeof text === 'string') {
+        if (isMarkdown) {
+            const html = marked(text, {
+                breaks: true,
+                sanitize: true,
+                renderer: new CustomRenderer(),
+            });
 
-        if (typeof text === 'object') {
-            return text;
+            return <span dangerouslySetInnerHTML={{__html: html}}/>;
         }
 
+        return <span>{text}</span>;
+    }
+
+    if ('id' in text) {
         if (isMarkdown) {
             return (
                 <FormattedMarkdownMessage
-                    id={text}
-                    defaultMessage={textDefault}
+                    {...text}
                     values={textValues}
                 />
             );
         }
 
         return (
             <FormattedMessage
-                id={text}
+                {...text}
                 values={textValues}
-                defaultMessage={textDefault}
             />
         );
-    };
-
-    renderUntranslated = () => {
-        const {isMarkdown, text} = this.props;
-        if (isMarkdown) {
-            if (typeof text === 'object') {
-                return text;
-            }
-            const html = marked(text, {
-                breaks: true,
-                sanitize: true,
-                renderer: new CustomRenderer(),
-            });
-
-            return <span dangerouslySetInnerHTML={{__html: html}}/>;
-        }
+    }
 
-        return <span>{text}</span>;
-    };
+    return text as JSX.Element;
```

Comment:
Ah, I see. I keep forgetting that `id` is optional in `MessageDescriptor` because we always define it in the web app code. This seems good enough to me then.